packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/record-position.factory.ts

```diff
@@ -19,40 +20,76 @@ export class RecordPositionFactory {
     value: number | 'first' | 'last',
     objectMetadata: { isCustom: boolean; nameSingular: string },
     workspaceId: string,
+    index = 0,
   ): Promise<number> {
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
     if (typeof value === 'number') {
+      const recordWithSamePosition = await this.findRecord(
+        {
+          recordPositionQueryType: RecordPositionQueryType.FIND_BY_POSITION,
+          positionValue: value,
+        },
+        objectMetadata,
+        dataSourceSchema,
+        workspaceId,
+      );
+
+      if (recordWithSamePosition) {
+        throw new Error('Position is not unique');
+      }
+
       return value;
     }
 
-    const dataSourceSchema =
-      this.workspaceDataSourceService.getSchemaName(workspaceId);
+    if (value === 'first') {
+      const firstRecord = await this.findRecord(
+        {
+          recordPositionQueryType: RecordPositionQueryType.FIND_FIRST_RECORD,
+        },
+        objectMetadata,
+        dataSourceSchema,
+        workspaceId,
+      );
+
+      return isDefined(firstRecord?.position)
+        ? firstRecord.position - index - 1
+        : 1;
+    }
+
+    const lastRecord = await this.findRecord(
+      {
+        recordPositionQueryType: RecordPositionQueryType.FIND_LAST_RECORD,
```

Comment:
Is this the only usage we have of FIND_LAST_RECORD - to get the maximum position? If so why not get the maximum of the position column? I think this is less costly (according to chatGPT):

`SELECT MAX(position) AS max_position FROM myTable;`

_Why MAX is More Efficient
Less Data Processing: MAX scans the column once to find the maximum value, which is generally faster than sorting all the rows.
Less Memory Usage: A query with ORDER BY may require storing and sorting all the values in memory, whereas MAX only needs a single pass through the data.
Index Optimization: If the position column is indexed, the MAX function can directly leverage this index, making the search even faster._

I don't know if we have a stake of performance here though

Reply:
Im also wondering what happens if some rows have a null position - is it possible? will this be an issue when fetching first or last record?

Reply:
This is a good approach. Checking on SO, I don't think it will be more efficient:
- I do not really do an `ORDER BY` that `require storing and sorting all the values in memory`. I do `ORDER BY ... LIMIT 1`. It means it will do the same as `MAX` or `MIN`, because it will stop once finding the first value to sort.
- A difference between both would be if we have `null` values. Then, for a sort, `null` may be sorted first, which would take more time than MIN. 

But it allows to only get the position + deal with remaining `null` values so I did the update.

Thank you for this comment, I learned something! 👌